sdk/keyvault/keyvault-keys/migration-guide.md

```diff
@@ -0,0 +1,257 @@
+
+To try and improve the development experience across Azure services, a set of uniform [design guidelines](https://azure.github.io/azure-sdk/general_introduction.html) was created for all languages to drive a consistent experience with established API patterns for all services. A set of [TypeScript specific guidelines][ts-guidelines] was also introduced to ensure that JavaScript clients have a natural and idiomatic feel with respect to the JavaScript and TypeScript ecosystems. Further details are available in the guidelines for those interested.
+
+### Cross Service SDK improvements
+
+The modern Key Vault client libraries also share some of the cross-service improvements made to the Azure development experience, such as:
+- Using the new `@azure/identity` library to share a single authentication approach between clients.
+- A unified logging and diagnostics pipeline that offers a common view of the activities across each of the client libraries.
+- (In case of JavaScript) The use of promises rather than callbacks for a simplified programming experience.
+- (In case of JavaScript) The use of async iterators in paging APIs.
+
+## Important changes
+
+### Separate packages and clients
+
+In the interest of simplifying the API `azure-keyvault` and `KeyVaultClient` were split into separate packages and clients:
+
+- [`@azure/keyvault-keys`][kvk-readme] contains `KeyClient` for working with Key Vault keys, and `CryptographyClient` for performing cryptographic operations.
+- [`@azure/keyvault-secrets`][kvs-readme] contains `SecretClient` for working with secrets.
+- [`@azure/keyvault-certificates`][kvc-readme] contains `CertificateClient` for working with certificates.
+
+### Client constructors
+
+Across all of the new Azure client libraries, clients consistently take an endpoint or connection string along with token credentials on their constructors. This differs from the legacy `KeyVaultClient`, which didn't receive the vault endpoint on the constructor, but required users to pass the vault endpoint as the first parameter to the methods that the `KeyVaultClient` provided.
+
+#### Authenticating
+
+Previously in `azure-keyvault` you could create a `KeyVaultClient` by using credentials from `ms-rest-azure` (up to the version `^2.6.0`. Higher versions are not supported).
+
+The dependencies on the `package.json` for a project with `azure-keyvault` would look like the following:
+
+```json
+  "dependencies": {
+    "azure-keyvault": "^3.0.5",
+    "ms-rest-azure": "^2.6.0"
+  }
+```
+
+With that `package.json` a simple use case of `azure-keyvault` would look as follows:
```

Comment:
I would recommend skipping this part. Since our focus is to move away from the older package, we don't need to go into details of how the package.json would look like to limit the version of ms-rest-azure
```suggestion
```